x-pack/solutions/security/plugins/security_solution/server/fleet_integration/handlers/install_elastic_defend.ts

```diff
@@ -0,0 +1,81 @@
+import type { SecuritySolutionApiRequestHandlerContext } from '../../types';
+import { ELASTIC_SECURITY_RULE_ID } from '../../../common';
+import { createPrebuiltRuleObjectsClient } from '../../lib/detection_engine/prebuilt_rules/logic/rule_objects/prebuilt_rule_objects_client';
+import { createPrebuiltRuleAssetsClient } from '../../lib/detection_engine/prebuilt_rules/logic/rule_assets/prebuilt_rule_assets_client';
+import { createPrebuiltRules } from '../../lib/detection_engine/prebuilt_rules/logic/rule_objects/create_prebuilt_rules';
+
+export interface InstallElasticDefendProps {
+  logger: Logger;
+  context: SecuritySolutionApiRequestHandlerContext;
+  request: KibanaRequest;
+  alerts: AlertingServerStart;
+  soClient: SavedObjectsClientContract;
+}
+
+/**
+ * As part of a user taking advantage of Endpoint Security from within fleet, we attempt to install
+ * the pre-packaged Elastic Defend Rule enabled by default
+ */
+export const installElasticDefend = async ({
+  logger,
+  context,
+  request,
+  alerts,
+  soClient,
+}: InstallElasticDefendProps): Promise<void> => {
+  // Create detection index & rules (if necessary). move past any failure, this is just a convenience
+  try {
+    await createDetectionIndex(context);
+  } catch (err) {
+    if (err.statusCode !== 409) {
+      // 409 -> detection index already exists, which is fine
+      logger.warn(
+        `Possible problem creating detection signals index (${err.statusCode}): ${err.message}`
+      );
+    }
+  }
+  try {
+    // this checks to make sure index exists first, safe to try in case of failure above
+    // may be able to recover from minor errors
+    const rulesClient = await alerts.getRulesClientWithRequest(request);
+    const detectionRulesClient = context.getDetectionRulesClient();
+    const ruleAssetsClient = createPrebuiltRuleAssetsClient(soClient);
+    const ruleObjectsClient = createPrebuiltRuleObjectsClient(rulesClient);
+
+    const elasticDefendRule = await ruleObjectsClient.fetchInstalledRulesByIds({
+      ruleIds: [ELASTIC_SECURITY_RULE_ID],
+    });
+    if (elasticDefendRule.length > 0) {
+      // Elastic Defend rule already installed
+      return;
+    }
+    // Elastic Defend rule not installed, find the latest version in the
+    // prebuilt rule assets and install it
+
+    const latestRuleVersion = await ruleAssetsClient.fetchLatestVersions([
+      ELASTIC_SECURITY_RULE_ID,
+    ]);
+    if (latestRuleVersion.length === 0) {
+      logger.error(
+        `Unable to find Elastic Defend rule in the prebuilt rule assets (rule_id: ${ELASTIC_SECURITY_RULE_ID})`
+      );
+      return;
+    }
+    const ruleAssetsToInstall = await ruleAssetsClient.fetchAssetsByVersion(latestRuleVersion);
+    await createPrebuiltRules(detectionRulesClient, ruleAssetsToInstall);
+  } catch (err) {
+    logger.error(
+      `Unable to create detection rules automatically (${err.statusCode}): ${err.message}`
+    );
+  }
```

Comment:
I think this logic should be on our side where our team will be the code owner for it.